GymApp/GYM.BLL.Tests/Services/CouchServiceTests.cs

```diff
@@ -0,0 +1,174 @@
+        public async Task Get_InputInvalidIdCheckNull_ReturnsNull(int id)
+        {
+            //Arrange
+            _repository.Setup(cr => cr.Get(It.IsAny<int>())).ReturnsAsync(TestEntities.GetCouchEntitiesForTest().FirstOrDefault(ce => ce.Id == id));
+
+            _mapperMoq.Setup(m => m.Map<CouchModel>(It.IsAny<CouchEntity>())).Returns(TestModels.GetCouchModelsForTest().FirstOrDefault(cm => cm.Id == id)!);
+
+            //Act
+            var resultCouchesModel = await _couchService.Get(id);
+
+            //Assert
+            resultCouchesModel.ShouldBeNull();
+        }
+
+        [Theory]
+        [InlineData("Arnold")]
+        [InlineData("Mick")]
+        [InlineData("Olivia")]
+        public async Task Get_InputPredicateCheckName_ReturnsCouchesModel(string expectedName)
+        {
+            //Arrange
+            Expression<Func<CouchEntity, bool>> predicateEntity = (pe) => pe.FirstName == expectedName;
+            Expression<Func<CouchModel, bool>> modelPredicate = (mp) => mp.FirstName == expectedName;
+
+            _repository.Setup(cr => cr.Get(It.IsAny<Expression<Func<CouchEntity, bool>>>())).ReturnsAsync(TestEntities.GetCouchEntitiesForTest());
+
+            _mapperMoq.Setup(m =>
+                    m.Map<Expression<Func<CouchEntity, bool>>>(It.IsAny<Expression<Func<CouchModel, bool>>>()))
+                .Returns(predicateEntity);
+
+            _mapperMoq.Setup(m => m.Map<IEnumerable<CouchModel>>(It.IsAny<IEnumerable<CouchEntity>>()
+            )).Returns(TestModels.GetCouchModelsForTest().Where(ce => ce.FirstName == expectedName));
+
+            _mapperMoq.Setup(m => m.Map<IEnumerable<CouchEntity>, IEnumerable<CouchModel>>(It.IsAny<IEnumerable<CouchEntity>>()))
+                .Returns(TestModels.GetCouchModelsForTest().Where(ce => ce.FirstName == expectedName));
+
+            //Act
+            var resultCouchesModel = await _couchService.Get(modelPredicate);
+
+            //Assert
+            resultCouchesModel.ShouldAllBe(c => c.FirstName == expectedName);
+        }
+
+        [Fact]
+        public async Task Create_InputCouchModelSaveCouchEntity_VerifyInvokeCreate()
+        {
+            //Arrange
+            var id = 1;
+            var couchEntity = TestEntities.GetCouchEntitiesForTest().FirstOrDefault(e => e.Id == id)!;
+            var couchModel = TestModels.GetCouchModelsForTest().FirstOrDefault(e => e.Id == id)!;
+
+            _mapperMoq.Setup(m => m.Map<CouchEntity>(It.IsAny<CouchModel>())).Returns(couchEntity);
+
+            //Act
+            await _couchService.Create(couchModel);
+
+            //Assert
+            _repository.Verify(p => p.Create(It.IsAny<CouchEntity>()), Times.Once);
+        }
+
+
+        [Fact]
+        public async Task Update_InputCouchModelUpdateCouchEntity_VerifyInvokeUpdate()
+        {
+            //Arrange
+            var id = 1;
+            var couchEntity = TestEntities.GetCouchEntitiesForTest().FirstOrDefault(e => e.Id == id)!;
```

Comment:
You shouldn't use shared data across multiple test cases, because, as your tests run in parallel and interact with this data simultaneously, it can lead to unpredictable mutations. In general, your test cases have to be isolated